lib/amazonka/src/Amazonka/EC2/Metadata.hs

```diff
@@ -295,7 +296,7 @@ isEC2 :: MonadIO m => Client.Manager -> m Bool
 isEC2 m = liftIO (Exception.catch req err)
   where
     req = do
-      !_ <- request m "http://instance-data/latest"
+      !_ <- get m "http://instance-data/latest"
```

Comment:
This will actually do an unnecessary request: as soon as we get a token back we are able to say "yes, we're on EC2". But I don't think it's worth fixing; raise another PR if you disagree?